lightning-graph-sync/README.md

```diff
@@ -0,0 +1,75 @@
+# lightning-graph-sync
+
+This crate exposes functionality for rapid gossip graph syncing, aimed primarily
+at mobile clients.
+
+## Mechanism
+
+The (presumed) server sends a compressed gossip response containing gossip data.
+The gossip data is formatted compactly, omitting signatures and opportunistically
+incremental.
+
+Essentially, the serialization structure is as follows:
+
+1. Fixed prefix bytes `76, 68, 75, 2` (the first three bytes are ASCII for `LDK`)
+    - The purpose of this prefix is to identify the serialization format, should other
+    rapid gossip sync formats arise in the future
+2. A `CompactSize` indicating the number of `ChannelAnnouncement` messages to follow
+3. `[ChannelAnnouncement]` (array of almost BOLT-7-compliant channel announcements, 
+    with the distinction that the signatures are stripped)
+4. A `CompactSize` indicating the number of channel update messages to follow`
+5. `[CustomChannelUpdate]`
+
+### CustomChannelUpdate
+
+For the purpose of rapid syncing, we have deviated from the channel update format
+specified in BOLT 7 significantly. Our custom channel updates are structured as follows:
+
+1. `short_channel_id`: `u64`
+2. `custom_channel_flags`: `u8`
+3. `update_data`
+
+Specifically, our custom channel flags break down like this:
+
+| 128                 | 64 | 32 | 16 | 8 | 4   | 2                | 1         |
```

Comment:
Is there some significance in the spacing difference in the columns for `8` and `4`?

Reply:
no, my IDE just decided to add it. I'll remove.